modules/messages/client/components/Inbox.component.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import * as api from '../api/messages.api';
+import Avatar from '@/modules/users/client/components/Avatar.component';
+import Activate from '@/modules/users/client/components/Activate';
+import { $broadcast, eventTrack } from '@/modules/core/client/services/angular-compat';
+import TimeAgo from '@/modules/core/client/components/TimeAgo';
+
+const userType = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
+  username: PropTypes.string.isRequired,
+  displayName: PropTypes.string.isRequired,
+});
+
+export default function Inbox({ user }) {
+  if (!user.public) {
+    return (
+      <section className="container-spacer">
+        <Activate/>
+      </section>
+    );
+  }
+  const { t } = useTranslation('messages');
+
+  const [nextParams, setNextParams] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+  const [threads, setThreads] = useState([]);
+
+  const hasMore = Boolean(nextParams);
+
+  async function fetchThreads(next = false) {
+    setIsFetching(true);
+    try {
+      if (next) {
+        eventTrack('inbox-pagination', {
+          category: 'messages.inbox',
+          label: 'Inbox page ' + nextParams,
+        });
+      }
+
+      const data = await api.fetchThreads(next ? nextParams : {});
+      setThreads(threads => next ? threads.concat(data.threads) : data.threads);
+      setNextParams(data.nextParams);
+    } finally {
+      setIsFetching(false);
+    }
+  }
+
+  useEffect(() => {
+    // Tell unread-messages directive to sync itself
+    $broadcast('syncUnreadMessagesCount');
+    fetchThreads();
+  }, []);
+
+  return <section className="container-spacer">
+    {!isFetching && threads.length === 0 && (
+      <div className="content-empty">
+        <i className="icon-3x icon-messages-alt"/>
+        <h4 role="alert">{t('No conversations yet.')}</h4>
+      </div>
+    )}
+    {threads.length > 0 && (
+      <ul className="list-group threadlist">
+        {threads.map(thread => (
+          <InboxThread
+            key={thread._id}
+            user={user}
+            thread={thread}
+          />
+        ))}
+      </ul>
+    )}
+    {!isFetching && hasMore && (
+      <div className="text-center">
+        <button
+          className="btn btn-primary btn-lg"
+          onClick={() => fetchThreads(true)}
+        >
+          {t('More messages')}
+        </button>
+      </div>
+    )}
+  </section>;
+}
+
+Inbox.propTypes = {
+  user: userType.isRequired,
+};
+
+function InboxThread({ user, thread }) {
```

Comment:
What do you think about splitting this component into a separate file for clarity? I like one-component-per-file approach quite a bit since it encourages re-using.

Not a blocker tho.

Reply:
One of the things I like about using react is a component is as lightweight as a function, so easy to split things off into microcomponents, which might not make sense to have in seperate files if they are tiny, in this case maybe it justifies a seperate component. I'll split it out.

Reply:
Definitely case by case. 👍

Reply:
I separated it out now :)